### DISCUSSION TITLE
Testing with multi bit input

### DISCUSSION THREAD

**Discussion by schauma** [hneemann/Digital#1033] [Q&A] [answered]
Hello,
I wanted to test an eight bit AND circuit and I just read about the test functionalities.
However I was wondering how I can write the test cases if my input is 8 bit and not 8 single bit inputs.
My attempt was the following, however it does not work and gives the error
"Value a is not a number!". Now the questions I have is
What is a if not a number?
How could this be done otherwise?

Thanks in advance :)

```
### 8 Bit AND circuit test

A B Y

loop(a,256)
	loop(b,256)
		a b ab
	end loop
end loop
```

**Comment by hneemann** [comment, marked as answer]
Since the parser for the test cases must also be able to parse a simple truth table, expressions must be enclosed in brackets.

```
A B Y

loop(a,256)
  loop(b,256)
     (a) (b) (a & b)
  end loop
end loop
```

**Reply by schauma** [reply]
Thank you!.
Works perfectly :)

As a follow up question. Is there a place where I can find all the possible commands for the parser? Does it follow the commands of another language perhaps? 
I am curious because maybe in the future I am want to test XOR/NOR etc shift commands etc.

**Reply by stephen-riley** [reply]
@hneemann  +1 to the question of any documentation about the test script language?
